subsys/net/l2/openthread/openthread.c

```diff
@@ -227,17 +239,75 @@ void ot_receive_handler(otMessage *aMessage, void *context)
 	otMessageFree(aMessage);
 }
 
+static struct openthread_context *ot_context_for_restarter;
+
+void ot_joiner_start_handler(otError error, void *context);
+
+static void ot_joiner_restarter(void *dummy)
+{
+	otError error = otJoinerStart(ot_context_for_restarter->instance,
+					joiner_pskd, NULL,
+					PACKAGE_NAME, OT_PLATFORM_INFO,
+					PACKAGE_VERSION, NULL,
+					&ot_joiner_start_handler,
+					ot_context_for_restarter);
+
+	if (error != OT_ERROR_NONE) {
+		NET_ERR("Failed to start joiner [%d]", error);
+	}
+}
+
+K_WORK_DEFINE(joiner_restart, ot_joiner_restarter);
+
+static void ot_joiner_restart_timer(void *dummy)
+{
+	k_work_submit(&joiner_restart);
+}
+
+K_TIMER_DEFINE(joiner_restart_timer, ot_joiner_restart_timer, NULL);
 void ot_joiner_start_handler(otError error, void *context)
 {
 	struct openthread_context *ot_context = context;
-
+	static u32_t restart_delay;
+	static u32_t join_attempts;
 	switch (error) {
 	case OT_ERROR_NONE:
 		NET_INFO("Join success");
 		otThreadSetEnabled(ot_context->instance, true);
 		break;
 	default:
 		NET_ERR("Join failed [%d]", error);
+		if (IS_ENABLED(CONFIG_OPENTHREAD_JOINER_CONTINUOUS)) {
+			join_attempts++;
+
+			if (CONFIG_OPENTHREAD_JOINER_REBOOT != 0) {
```

Comment:
It would be more concise to join the two `if` cases and use `&&` operator.

Reply:
It was meant that with CONFIG_OPENTHREAD_JOINER_CONTINUOUS we can activate the joiner loop. In the joiner loop there are 3 more configurable options:

1. - CONFIG_OPENTHREAD_JOINER_ADD_DELAY_MS: Added joining delay [ms] till next try.
2. - CONFIG_OPENTHREAD_JOINER_TRIALS_MAX_DELAY: Number of joining trials till the maximum delay is reached. If it is configured with 0, there is no maxmium delay.
3. - CONFIG_OPENTHREAD_JOINER_REBOOT: Reboots the device after a specified number of failed joining trials. If it is configured with 0, it never reboots.

That's why I made a few if statements in another if statement. With CONFIG_OPENTHREAD_JOINER_CONTINUOUS && CONFIG_OPENTHREAD_JOINER_REBOOT  these two would basically be the same option and there is no possibility to start the joining loop in an endless loop.

Reply:
I understand the overall structure of this joiner loop, my comment only referred to the line we're commenting on, i.e. to have:
```
if ((CONFIG_OPENTHREAD_JOINER_REBOOT != 0) &&
    (join_attempts >= CONFIG_OPENTHREAD_JOINER_REBOOT)) {
```
but it's not a big deal really, just one less indentation level - less line wrapping.

Reply:
Sorry my mistake I  looked at if (IS_ENABLED(CONFIG_OPENTHREAD_JOINER_CONTINUOUS) as second if. Of course I will make one if out of these 2.